server/server.js

```diff
@@ -1,35 +1,212 @@
 const express = require('express');
-const knex = require('./knex');  
-const axios = require('axios');  
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const cookieParser = require('cookie-parser');
 const cors = require('cors');
-const app = express();
-
-require('dotenv').config();
+const axios = require('axios'); 
+const knex = require('./knex'); 
+require('dotenv').config(); 
 
+const app = express();
 app.use(cors());
 app.use(express.json());
+app.use(cookieParser());
+
+
+function authenticateToken(req, res, next) {
+    const token = req.cookies.token;
+
+    if (!token) {
+        return res.status(401).json({ error: 'Access denied. No token provided.' });
+    }
+
+    jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+        if (err) {
+            return res.status(403).json({ error: 'Invalid token.' });
+        }
+        req.user = user;
+        next();
+    });
+}
 
+
+app.post('/register', async (req, res) => {
+    const { username, password } = req.body;
+
+    if (!username || !password) {
+        return res.status(400).json({ error: 'Username and password are required' });
+    }
+
+    try {
+        const hashedPassword = await bcrypt.hash(password, 10);
+        await knex('user').insert({
+            username: username,
+            password: hashedPassword
+        });
+
+        return res.status(201).json({ message: 'User registered successfully' });
+    } catch (error) {
+        console.log(error);
+        return res.status(500).json({ error: 'Error registering user' });
+    }
+});
+
+
+app.post('/login', async (req, res) => {
+    const { username, password } = req.body;
+
+    if (!username || !password) {
+        return res.status(400).json({ error: 'Username and password are required' });
+    }
+
+    try {
+        const user = await knex('user').where({ username }).first();
+        if (!user) {
+            return res.status(400).json({ error: 'User not found' });
+        }
+
+        const isMatch = await bcrypt.compare(password, user.password);
+        if (!isMatch) {
+            return res.status(400).json({ error: 'Invalid password' });
+        }
+
+        const token = jwt.sign({ username: user.username }, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '1h' });
+        res.cookie('token', token, { httpOnly: true, secure: true, maxAge: 3600000 });
+
+        return res.status(200).json({ message: 'Login successful' });
+    } catch (error) {
+        return res.status(500).json({ error: 'Error during login' });
+    }
+});
+
```

Comment:
Oh 😮 I'm a little surprised to see this.

You asked me to work on authorization this morning, so I opened an issue and I spent the morning writing the login and the register functionality. I told you I was almost finished by lunch. Then after lunch I tried to ask you if we could combine our code, but you said we were wasting time talking.

Sorry to say, I feel like I wasted a whole morning writing the authorization part if you already wrote this and I didn't know about it 😕.

I think we need to make the division of our tasks clearer in the morning and stop rushing into coding.